Pokedex_ios/View/Controller/PokemonList/PokemonListViewController.swift

```diff
@@ -26,26 +26,39 @@ class PokemonListViewController: UIViewController {
         setBackground()
         initViewModel()
         setSearchView()
-        // Setup for the advanced filter image and button
-        setUpAdvanceFilterPopup()
+        // Setup for all filter image and button
+        setFiltersIconsPopup()
         // Register the custom cell
         setSearchView()
         let uiNib = UINib(nibName: pokemonCell, bundle: nil)
         pokemonsTableView.register(uiNib, forCellReuseIdentifier: pokeCellIdentifier)
     }
-    func setUpAdvanceFilterPopup() {
+    func setFiltersIconsPopup() {
         let advancedFilterImage = UIImage(named: "AdvancedFilterSVG")?.withRenderingMode(.alwaysOriginal)
         let advancedFilterButton = UIBarButtonItem(image: advancedFilterImage, style: .plain,
                                                    target: self, action: #selector(showAdvancedFilterPopup))
-        navigationItem.rightBarButtonItem = advancedFilterButton
+
+        let generationFilterImage = UIImage(named: "generationFilterIcon")?.withRenderingMode(.alwaysOriginal)
```

Comment:
The problem wasnt the method name here, it was better that you create your own method to do your stuff, not reuse the one that it was